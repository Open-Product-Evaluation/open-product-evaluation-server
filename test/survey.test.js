// TODO SurveyUpdate ReorderQuestions
// TODO SurveyUpdate Schould not update public surveys


const users = require('../seeds/data/user/user')
const surveys = require('../seeds/data/survey/survey')
const contexts = require('../seeds/data/context/context')
const { seedDatabase } = require('mongo-seeding')
const config = require('../config')
const request = require('./requesthelper')
const { getSeedID } = require('./helpers')

/* Functions for Querys */

function createSurveyQuery(surveyTitle, surveyDescription, isPublic) {
  return {
    query: `mutation {
      createSurvey(data: {title: "${surveyTitle}", description: "${surveyDescription}", isPublic: ${isPublic}}) {
        survey {
          creator{
            firstName
          }
          title
          description
          isPublic
          types
          questions {
            id
          }
          votes {
            id
          }
          contexts {
            id
          }
          images {
            id
          }
        }
      }
    }`,
  }
}

function updateSurveyQuery(surveyID, surveyTitle, surveyDescription, isPublic) {
  return {
    query: `mutation {
      updateSurvey(surveyID: "${surveyID}", data: {title: "${surveyTitle}", description: "${surveyDescription}", isPublic: ${isPublic}}) {
        survey {
          creator{
            firstName
          }
          title
          description
          isPublic
          types
          questions {
            id
          }
          votes {
            id
          }
          contexts {
            id
          }
          images {
            id
          }
        }
      }
    }`,
  }
}

function deleteSurveyQuery(surveyID) {
  return {
    query: `mutation {
      deleteSurvey(surveyID:"${surveyID}"){
        success
      }
    }`,
  }
}

function surveysQuery() {
  return {
    query: `{
      surveys {
        creator{
          firstName
        }
        title
        description
        isPublic
        types
        questions {
          id
        }
        votes {
          id
        }
        contexts {
          id
        }
        images {
          id
        }
      }
    }`,
  }
}

function surveyQuery(surveyID) {
  return {
    query: `{
      survey(surveyID:"${surveyID}"){
        creator{
          firstName
        }
        title
        description
        isPublic
        types
        questions {
          id
        }
        votes {
          id
        }
        contexts {
          id
        }
        images {
          id
        }
      }
    }`,
  }
}

/* Tests */

describe('Survey', () => {
  describe('Admin', async () => {
    let jwtToken = ''
    beforeAll(async () => {
      await seedDatabase(config.seeder)
      const expected = users[1]
      const query = {
        query: `mutation {
          login(data: {email: "${expected.email}", password: "password"}) {
            token
            user {
              id
              firstName
              lastName
              email
            }
          }
        }`,
      }
      const { data, errors } = await request.anon(query)
      data.login.token.should.be.a('string')
      expect(errors).toBeUndefined()
      const { login: { token } } = data
      jwtToken = token
    })
    it('schould return all surveys [Query]', async () => {
      const query = surveysQuery()
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould return survey owned by User [Query]', async () => {
      const survey = surveys[1]
      const query = surveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould return survey not owned by User [Query]', async () => {
      const survey = surveys[0]
      const query = surveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould create survey [Mutation]', async () => {
      const query = createSurveyQuery('TestSurvey', 'TestDescription', false)
      const res = await request.user(query, jwtToken)
      const { data, errors } = res
      expect(data).toMatchSnapshot()
      expect(errors).toBeUndefined()
    })
    it('schould update survey owned by User [Mutation]', async () => {
      const survey = surveys[1]
      const query = updateSurveyQuery(getSeedID(survey), 'RenamedTestSurvey', 'UpdatedSurveyDescription', true)
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould update survey not owned by User [Mutation]', async () => {
      const survey = surveys[0]
      const query = updateSurveyQuery(getSeedID(survey), 'RenamedTestSurvey', 'UpdatedSurveyDescription', true)
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould delete survey owned by user [Mutation]', async () => {
      const survey = surveys[1]
      const query = deleteSurveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data.deleteSurvey.success).toBe(true)
    })
    it('schould delete survey not owned by user [Mutation]', async () => {
      const survey = surveys[0]
      const query = deleteSurveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data.deleteSurvey.success).toBe(true)
    })
  })
  describe('User', async () => {
    let jwtToken = ''
    beforeAll(async () => {
      await seedDatabase(config.seeder)
      const expected = users[0]
      const query = {
        query: `mutation {
          login(data: {email: "${expected.email}", password: "password"}) {
            token
            user {
              id
              firstName
              lastName
              email
            }
          }
        }`,
      }
      const { data, errors } = await request.anon(query)
      data.login.token.should.be.a('string')
      expect(errors).toBeUndefined()
      const { login: { token } } = data
      jwtToken = token
    })
    it('schould return all surveys [Query]', async () => {
      const query = surveysQuery()
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould return survey owned by User [Query]', async () => {
      const survey = surveys[0]
      const query = surveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould not return survey not owned by User [Query]', async () => {
      const survey = surveys[1]
      const query = surveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('schould create survey [Mutation]', async () => {
      const query = createSurveyQuery('TestSurvey', 'TestDescription', false)
      const res = await request.user(query, jwtToken)
      const { data, errors } = res
      expect(data).toMatchSnapshot()
      expect(errors).toBeUndefined()
    })
    it('schould update survey owned by User [Mutation]', async () => {
      const survey = surveys[0]
      const query = updateSurveyQuery(getSeedID(survey), 'RenamedTestSurvey', 'UpdatedSurveyDescription', true)
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data).toMatchSnapshot()
    })
    it('schould not update survey not owned by User [Mutation]', async () => {
      const survey = surveys[1]
      const query = updateSurveyQuery(getSeedID(survey), 'RenamedTestSurvey', 'UpdatedSurveyDescription', true)
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('schould delete survey owned by user [Mutation]', async () => {
      const survey = surveys[0]
      const query = deleteSurveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(errors).toBeUndefined()
      expect(data.deleteSurvey.success).toBe(true)
    })
    it('schould not delete survey not owned by user [Mutation]', async () => {
      const survey = surveys[1]
      const query = deleteSurveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
  })
  describe.skip('Device', async () => {
    let jwtToken = ''
    beforeAll(async () => {
      await seedDatabase(config.seeder)
      const query = {
        query: `mutation{createDevice(data:{name:"TestDevice"}){
          token
       }}`,
      }
      const { data, errors } = await request.anon(query)
      data.createDevice.token.should.be.a('string')
      expect(errors).toBeUndefined()
      const { createDevice: { token } } = data
      jwtToken = token
    })
    it('schould not return all surveys [Query]', async () => {
      const query = surveysQuery()
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('schould not return survey owned by User [Query]', async () => {
      const survey = surveys[0]
      const query = surveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('schould update survey [Mutation]', async () => {
      const survey = surveys[0]
      const context = contexts[0]
      const user = users[0]
      const query = updateSurveyQuery(getSeedID(survey), 'RenamedTestSurvey', getSeedID(context), [getSeedID(user)])
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
    it('schould not delete survey [Mutation]', async () => {
      const survey = surveys[3]
      const query = deleteSurveyQuery(getSeedID(survey))
      const { data, errors } = await request.user(query, jwtToken)
      expect(data).toBeNull()
      expect(errors.length).toBeGreaterThan(0)
    })
  })
})
