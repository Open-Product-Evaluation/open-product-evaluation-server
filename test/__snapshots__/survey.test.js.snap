// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Survey Admin should create survey [Mutation] 1`] = `
Object {
  "createSurvey": Object {
    "survey": Object {
      "creator": Object {
        "firstName": "Jane",
      },
      "description": "TestDescription",
      "domains": null,
      "images": null,
      "isActive": false,
      "questions": null,
      "title": "TestSurvey",
      "types": null,
      "votes": null,
    },
  },
}
`;

exports[`Survey Admin should return all surveys [Query] 1`] = `
Object {
  "surveys": Array [
    Object {
      "creator": Object {
        "firstName": "John",
      },
      "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Wie verhalten sich Informatiker?\\"",
      "domains": Array [
        Object {
          "id": "d8134f182c41d3cf4e9fdcda2be39855d7f1530757f06392c615fe652edee1a2",
        },
      ],
      "images": Array [
        Object {
          "id": "764b22f9fe765b1df73d443d578f5dfcda7a53040ee1dcbcc03a9c1fc3089673",
        },
        Object {
          "id": "53ea1fb253f463f072bbb075c62998ae78b5cacb46d9fe9a8456cc5db67038e0",
        },
      ],
      "isActive": false,
      "questions": Array [
        Object {
          "id": "c237a33fc025101813b9bb9fbafe22fd9d53dbd110f48cec951fbeaf34ff3622",
        },
        Object {
          "id": "50ffe7d198c0f5dabda06fbb6db3ed4751eb2a1eb8032ac65bea6dbd4d68150a",
        },
        Object {
          "id": "b3c6a227951ed7b707222be79338c50c41ab7e51213551aabe48e0ee2148cbb6",
        },
        Object {
          "id": "534c13385adba70a40f9b898396e959e668dd1e0dc1e7de4c61c14d668965184",
        },
        Object {
          "id": "66ccc956897f52437656ba7e5588872baf213035f3a47a8bbae2b51289653078",
        },
        Object {
          "id": "2e26805e909bd52bf6aa07d4edce45575d6ba6f56e32b71f7466cda92f3ab76e",
        },
      ],
      "title": "Untersuchung zum Verhalten von Informatikern in Hochschulen",
      "types": Array [
        "CHOICE",
        "LIKE",
        "LIKEDISLIKE",
        "REGULATOR",
        "FAVORITE",
        "RANKING",
      ],
      "votes": Array [
        Object {
          "id": "868d55aea73544110bd9e55daac7b585a3088e0ddfd7b0fcc50a864b342a19a4",
        },
        Object {
          "id": "d3292d8dbc87f3f8e78de5deec9df47bcf84cfb4caa2a2e622701394854c3ac5",
        },
      ],
    },
    Object {
      "creator": Object {
        "firstName": "Jane",
      },
      "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Künstliche Intelligenz?",
      "domains": Array [
        Object {
          "id": "624497235d090cf74bd2473155d6d65fb9f9395ec018192d181387e20f4cbf10",
        },
      ],
      "images": Array [
        Object {
          "id": "bd9cbdb811c9bade42f61908308847265d0ce0ae209f6f9ff2981f6c189f192d",
        },
        Object {
          "id": "5848ae29acabc3e1f99a86e8d7d76f6ce43cd362b10035e559b9a63d2e26fca5",
        },
        Object {
          "id": "31508b27871f0315ffc2d37269eb68faf2e9274290f23492f74a99f6f4b3d47f",
        },
        Object {
          "id": "df4d4fdf8abdf7727e04d7af5e65dc8b539bc16da9c1d84567d026c56cdc0850",
        },
      ],
      "isActive": false,
      "questions": Array [
        Object {
          "id": "fd0dca1800e20d92ed5e3c01fafbd5f45b26f57d6e5118506eaf9eca7407e833",
        },
        Object {
          "id": "e657e852e504aceed834f2a001b0e5f0190e4f84730906990c6e6207769db48a",
        },
        Object {
          "id": "77bda58fad59e0b68a58f7b92cd4be87618046657ed45a965b23091d39203b2b",
        },
        Object {
          "id": "51593cf78e9828f700aa86f2d400e0497676610be878089d2268b1715d2ce980",
        },
        Object {
          "id": "0db66302ab707ff4cb22d52c7bf09fc238713dbcc9f0998c839162671df9c85b",
        },
        Object {
          "id": "b49c447e06d23454ba1b365c8884e8ca363575ae671dca4eb7fd78eaa3eca54e",
        },
      ],
      "title": "Öffentliche Umfrage im Rahmen des Modules \\"Computerethik\\"",
      "types": Array [
        "FAVORITE",
        "LIKE",
        "LIKEDISLIKE",
        "RANKING",
      ],
      "votes": Array [
        Object {
          "id": "b527daa8b46239146457fc968fd45d4a9de789b71d31e3a9d1bce356f5f2a887",
        },
        Object {
          "id": "5c965165123016a8a9c39d45082ab2a2b6fed162daefeeba75e3d3bdbf9fbc2e",
        },
      ],
    },
    Object {
      "creator": Object {
        "firstName": "Jake",
      },
      "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Künstliche Intelligenz?",
      "domains": Array [
        Object {
          "id": "9f3b5b94695bd9255db28ec93e5114ca0e6ccf7af668ca086af18c9b3a8eaa63",
        },
      ],
      "images": null,
      "isActive": true,
      "questions": null,
      "title": "Öffentliche Umfrage im Rahmen des Modules \\"Computerethik (Kopie)\\"",
      "types": Array [
        "FAVORITE",
        "LIKE",
        "LIKEDISLIKE",
        "RANKING",
      ],
      "votes": null,
    },
  ],
}
`;

exports[`Survey Admin should return survey not owned by User [Query] 1`] = `
Object {
  "survey": Object {
    "creator": Object {
      "firstName": "John",
    },
    "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Wie verhalten sich Informatiker?\\"",
    "domains": Array [
      Object {
        "id": "d8134f182c41d3cf4e9fdcda2be39855d7f1530757f06392c615fe652edee1a2",
      },
    ],
    "images": Array [
      Object {
        "id": "764b22f9fe765b1df73d443d578f5dfcda7a53040ee1dcbcc03a9c1fc3089673",
      },
      Object {
        "id": "53ea1fb253f463f072bbb075c62998ae78b5cacb46d9fe9a8456cc5db67038e0",
      },
    ],
    "isActive": false,
    "questions": Array [
      Object {
        "id": "c237a33fc025101813b9bb9fbafe22fd9d53dbd110f48cec951fbeaf34ff3622",
      },
      Object {
        "id": "50ffe7d198c0f5dabda06fbb6db3ed4751eb2a1eb8032ac65bea6dbd4d68150a",
      },
      Object {
        "id": "b3c6a227951ed7b707222be79338c50c41ab7e51213551aabe48e0ee2148cbb6",
      },
      Object {
        "id": "534c13385adba70a40f9b898396e959e668dd1e0dc1e7de4c61c14d668965184",
      },
      Object {
        "id": "66ccc956897f52437656ba7e5588872baf213035f3a47a8bbae2b51289653078",
      },
      Object {
        "id": "2e26805e909bd52bf6aa07d4edce45575d6ba6f56e32b71f7466cda92f3ab76e",
      },
    ],
    "title": "Untersuchung zum Verhalten von Informatikern in Hochschulen",
    "types": Array [
      "CHOICE",
      "LIKE",
      "LIKEDISLIKE",
      "REGULATOR",
      "FAVORITE",
      "RANKING",
    ],
    "votes": Array [
      Object {
        "id": "868d55aea73544110bd9e55daac7b585a3088e0ddfd7b0fcc50a864b342a19a4",
      },
      Object {
        "id": "d3292d8dbc87f3f8e78de5deec9df47bcf84cfb4caa2a2e622701394854c3ac5",
      },
    ],
  },
}
`;

exports[`Survey Admin should return survey owned by User [Query] 1`] = `
Object {
  "survey": Object {
    "creator": Object {
      "firstName": "Jane",
    },
    "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Künstliche Intelligenz?",
    "domains": Array [
      Object {
        "id": "624497235d090cf74bd2473155d6d65fb9f9395ec018192d181387e20f4cbf10",
      },
    ],
    "images": Array [
      Object {
        "id": "bd9cbdb811c9bade42f61908308847265d0ce0ae209f6f9ff2981f6c189f192d",
      },
      Object {
        "id": "5848ae29acabc3e1f99a86e8d7d76f6ce43cd362b10035e559b9a63d2e26fca5",
      },
      Object {
        "id": "31508b27871f0315ffc2d37269eb68faf2e9274290f23492f74a99f6f4b3d47f",
      },
      Object {
        "id": "df4d4fdf8abdf7727e04d7af5e65dc8b539bc16da9c1d84567d026c56cdc0850",
      },
    ],
    "isActive": false,
    "questions": Array [
      Object {
        "id": "fd0dca1800e20d92ed5e3c01fafbd5f45b26f57d6e5118506eaf9eca7407e833",
      },
      Object {
        "id": "e657e852e504aceed834f2a001b0e5f0190e4f84730906990c6e6207769db48a",
      },
      Object {
        "id": "77bda58fad59e0b68a58f7b92cd4be87618046657ed45a965b23091d39203b2b",
      },
      Object {
        "id": "51593cf78e9828f700aa86f2d400e0497676610be878089d2268b1715d2ce980",
      },
      Object {
        "id": "0db66302ab707ff4cb22d52c7bf09fc238713dbcc9f0998c839162671df9c85b",
      },
      Object {
        "id": "b49c447e06d23454ba1b365c8884e8ca363575ae671dca4eb7fd78eaa3eca54e",
      },
    ],
    "title": "Öffentliche Umfrage im Rahmen des Modules \\"Computerethik\\"",
    "types": Array [
      "FAVORITE",
      "LIKE",
      "LIKEDISLIKE",
      "RANKING",
    ],
    "votes": Array [
      Object {
        "id": "b527daa8b46239146457fc968fd45d4a9de789b71d31e3a9d1bce356f5f2a887",
      },
      Object {
        "id": "5c965165123016a8a9c39d45082ab2a2b6fed162daefeeba75e3d3bdbf9fbc2e",
      },
    ],
  },
}
`;

exports[`Survey Admin should update survey not owned by User [Mutation] 1`] = `
Object {
  "updateSurvey": Object {
    "survey": Object {
      "creator": Object {
        "firstName": "John",
      },
      "description": "UpdatedSurveyDescription",
      "domains": Array [
        Object {
          "id": "d8134f182c41d3cf4e9fdcda2be39855d7f1530757f06392c615fe652edee1a2",
        },
      ],
      "images": Array [
        Object {
          "id": "764b22f9fe765b1df73d443d578f5dfcda7a53040ee1dcbcc03a9c1fc3089673",
        },
        Object {
          "id": "53ea1fb253f463f072bbb075c62998ae78b5cacb46d9fe9a8456cc5db67038e0",
        },
      ],
      "isActive": true,
      "questions": Array [
        Object {
          "id": "c237a33fc025101813b9bb9fbafe22fd9d53dbd110f48cec951fbeaf34ff3622",
        },
        Object {
          "id": "50ffe7d198c0f5dabda06fbb6db3ed4751eb2a1eb8032ac65bea6dbd4d68150a",
        },
        Object {
          "id": "b3c6a227951ed7b707222be79338c50c41ab7e51213551aabe48e0ee2148cbb6",
        },
        Object {
          "id": "534c13385adba70a40f9b898396e959e668dd1e0dc1e7de4c61c14d668965184",
        },
        Object {
          "id": "66ccc956897f52437656ba7e5588872baf213035f3a47a8bbae2b51289653078",
        },
        Object {
          "id": "2e26805e909bd52bf6aa07d4edce45575d6ba6f56e32b71f7466cda92f3ab76e",
        },
      ],
      "title": "RenamedTestSurvey",
      "types": Array [
        "CHOICE",
        "LIKE",
        "LIKEDISLIKE",
        "REGULATOR",
        "FAVORITE",
        "RANKING",
      ],
      "votes": Array [
        Object {
          "id": "868d55aea73544110bd9e55daac7b585a3088e0ddfd7b0fcc50a864b342a19a4",
        },
        Object {
          "id": "d3292d8dbc87f3f8e78de5deec9df47bcf84cfb4caa2a2e622701394854c3ac5",
        },
      ],
    },
  },
}
`;

exports[`Survey Admin should update survey owned by User [Mutation] 1`] = `
Object {
  "updateSurvey": Object {
    "survey": Object {
      "creator": Object {
        "firstName": "Jane",
      },
      "description": "UpdatedSurveyDescription",
      "domains": Array [
        Object {
          "id": "624497235d090cf74bd2473155d6d65fb9f9395ec018192d181387e20f4cbf10",
        },
      ],
      "images": Array [
        Object {
          "id": "bd9cbdb811c9bade42f61908308847265d0ce0ae209f6f9ff2981f6c189f192d",
        },
        Object {
          "id": "5848ae29acabc3e1f99a86e8d7d76f6ce43cd362b10035e559b9a63d2e26fca5",
        },
        Object {
          "id": "31508b27871f0315ffc2d37269eb68faf2e9274290f23492f74a99f6f4b3d47f",
        },
        Object {
          "id": "df4d4fdf8abdf7727e04d7af5e65dc8b539bc16da9c1d84567d026c56cdc0850",
        },
      ],
      "isActive": true,
      "questions": Array [
        Object {
          "id": "fd0dca1800e20d92ed5e3c01fafbd5f45b26f57d6e5118506eaf9eca7407e833",
        },
        Object {
          "id": "e657e852e504aceed834f2a001b0e5f0190e4f84730906990c6e6207769db48a",
        },
        Object {
          "id": "77bda58fad59e0b68a58f7b92cd4be87618046657ed45a965b23091d39203b2b",
        },
        Object {
          "id": "51593cf78e9828f700aa86f2d400e0497676610be878089d2268b1715d2ce980",
        },
        Object {
          "id": "0db66302ab707ff4cb22d52c7bf09fc238713dbcc9f0998c839162671df9c85b",
        },
        Object {
          "id": "b49c447e06d23454ba1b365c8884e8ca363575ae671dca4eb7fd78eaa3eca54e",
        },
      ],
      "title": "RenamedTestSurvey",
      "types": Array [
        "FAVORITE",
        "LIKE",
        "LIKEDISLIKE",
        "RANKING",
      ],
      "votes": Array [
        Object {
          "id": "b527daa8b46239146457fc968fd45d4a9de789b71d31e3a9d1bce356f5f2a887",
        },
        Object {
          "id": "5c965165123016a8a9c39d45082ab2a2b6fed162daefeeba75e3d3bdbf9fbc2e",
        },
      ],
    },
  },
}
`;

exports[`Survey User should create survey [Mutation] 1`] = `
Object {
  "createSurvey": Object {
    "survey": Object {
      "creator": Object {
        "firstName": "John",
      },
      "description": "TestDescription",
      "domains": null,
      "images": null,
      "isActive": false,
      "questions": null,
      "title": "TestSurvey",
      "types": null,
      "votes": null,
    },
  },
}
`;

exports[`Survey User should return all surveys [Query] 1`] = `
Object {
  "surveys": Array [
    Object {
      "creator": Object {
        "firstName": "John",
      },
      "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Wie verhalten sich Informatiker?\\"",
      "domains": Array [
        Object {
          "id": "d8134f182c41d3cf4e9fdcda2be39855d7f1530757f06392c615fe652edee1a2",
        },
      ],
      "images": Array [
        Object {
          "id": "764b22f9fe765b1df73d443d578f5dfcda7a53040ee1dcbcc03a9c1fc3089673",
        },
        Object {
          "id": "53ea1fb253f463f072bbb075c62998ae78b5cacb46d9fe9a8456cc5db67038e0",
        },
      ],
      "isActive": false,
      "questions": Array [
        Object {
          "id": "c237a33fc025101813b9bb9fbafe22fd9d53dbd110f48cec951fbeaf34ff3622",
        },
        Object {
          "id": "50ffe7d198c0f5dabda06fbb6db3ed4751eb2a1eb8032ac65bea6dbd4d68150a",
        },
        Object {
          "id": "b3c6a227951ed7b707222be79338c50c41ab7e51213551aabe48e0ee2148cbb6",
        },
        Object {
          "id": "534c13385adba70a40f9b898396e959e668dd1e0dc1e7de4c61c14d668965184",
        },
        Object {
          "id": "66ccc956897f52437656ba7e5588872baf213035f3a47a8bbae2b51289653078",
        },
        Object {
          "id": "2e26805e909bd52bf6aa07d4edce45575d6ba6f56e32b71f7466cda92f3ab76e",
        },
      ],
      "title": "Untersuchung zum Verhalten von Informatikern in Hochschulen",
      "types": Array [
        "CHOICE",
        "LIKE",
        "LIKEDISLIKE",
        "REGULATOR",
        "FAVORITE",
        "RANKING",
      ],
      "votes": Array [
        Object {
          "id": "868d55aea73544110bd9e55daac7b585a3088e0ddfd7b0fcc50a864b342a19a4",
        },
        Object {
          "id": "d3292d8dbc87f3f8e78de5deec9df47bcf84cfb4caa2a2e622701394854c3ac5",
        },
      ],
    },
  ],
}
`;

exports[`Survey User should return survey owned by User [Query] 1`] = `
Object {
  "survey": Object {
    "creator": Object {
      "firstName": "John",
    },
    "description": "Eine wissenschaftliche Umfrage mit der Forschungsfrage: \\"Wie verhalten sich Informatiker?\\"",
    "domains": Array [
      Object {
        "id": "d8134f182c41d3cf4e9fdcda2be39855d7f1530757f06392c615fe652edee1a2",
      },
    ],
    "images": Array [
      Object {
        "id": "764b22f9fe765b1df73d443d578f5dfcda7a53040ee1dcbcc03a9c1fc3089673",
      },
      Object {
        "id": "53ea1fb253f463f072bbb075c62998ae78b5cacb46d9fe9a8456cc5db67038e0",
      },
    ],
    "isActive": false,
    "questions": Array [
      Object {
        "id": "c237a33fc025101813b9bb9fbafe22fd9d53dbd110f48cec951fbeaf34ff3622",
      },
      Object {
        "id": "50ffe7d198c0f5dabda06fbb6db3ed4751eb2a1eb8032ac65bea6dbd4d68150a",
      },
      Object {
        "id": "b3c6a227951ed7b707222be79338c50c41ab7e51213551aabe48e0ee2148cbb6",
      },
      Object {
        "id": "534c13385adba70a40f9b898396e959e668dd1e0dc1e7de4c61c14d668965184",
      },
      Object {
        "id": "66ccc956897f52437656ba7e5588872baf213035f3a47a8bbae2b51289653078",
      },
      Object {
        "id": "2e26805e909bd52bf6aa07d4edce45575d6ba6f56e32b71f7466cda92f3ab76e",
      },
    ],
    "title": "Untersuchung zum Verhalten von Informatikern in Hochschulen",
    "types": Array [
      "CHOICE",
      "LIKE",
      "LIKEDISLIKE",
      "REGULATOR",
      "FAVORITE",
      "RANKING",
    ],
    "votes": Array [
      Object {
        "id": "868d55aea73544110bd9e55daac7b585a3088e0ddfd7b0fcc50a864b342a19a4",
      },
      Object {
        "id": "d3292d8dbc87f3f8e78de5deec9df47bcf84cfb4caa2a2e622701394854c3ac5",
      },
    ],
  },
}
`;

exports[`Survey User should update survey owned by User [Mutation] 1`] = `
Object {
  "updateSurvey": Object {
    "survey": Object {
      "creator": Object {
        "firstName": "John",
      },
      "description": "UpdatedSurveyDescription",
      "domains": Array [
        Object {
          "id": "d8134f182c41d3cf4e9fdcda2be39855d7f1530757f06392c615fe652edee1a2",
        },
      ],
      "images": Array [
        Object {
          "id": "764b22f9fe765b1df73d443d578f5dfcda7a53040ee1dcbcc03a9c1fc3089673",
        },
        Object {
          "id": "53ea1fb253f463f072bbb075c62998ae78b5cacb46d9fe9a8456cc5db67038e0",
        },
      ],
      "isActive": true,
      "questions": Array [
        Object {
          "id": "c237a33fc025101813b9bb9fbafe22fd9d53dbd110f48cec951fbeaf34ff3622",
        },
        Object {
          "id": "50ffe7d198c0f5dabda06fbb6db3ed4751eb2a1eb8032ac65bea6dbd4d68150a",
        },
        Object {
          "id": "b3c6a227951ed7b707222be79338c50c41ab7e51213551aabe48e0ee2148cbb6",
        },
        Object {
          "id": "534c13385adba70a40f9b898396e959e668dd1e0dc1e7de4c61c14d668965184",
        },
        Object {
          "id": "66ccc956897f52437656ba7e5588872baf213035f3a47a8bbae2b51289653078",
        },
        Object {
          "id": "2e26805e909bd52bf6aa07d4edce45575d6ba6f56e32b71f7466cda92f3ab76e",
        },
      ],
      "title": "RenamedTestSurvey",
      "types": Array [
        "CHOICE",
        "LIKE",
        "LIKEDISLIKE",
        "REGULATOR",
        "FAVORITE",
        "RANKING",
      ],
      "votes": Array [
        Object {
          "id": "868d55aea73544110bd9e55daac7b585a3088e0ddfd7b0fcc50a864b342a19a4",
        },
        Object {
          "id": "d3292d8dbc87f3f8e78de5deec9df47bcf84cfb4caa2a2e622701394854c3ac5",
        },
      ],
    },
  },
}
`;
